packages/runtime/runtime-utils/src/test/serializer.spec.ts

```diff
@@ -49,19 +71,63 @@ describe("FluidSerializer", () => {
                 assert.strictEqual(actual, input,
                     "replaceHandles() on input with no handles must return original input.");
 
+                const decoded = serializer.decode(actual);
+                assert.strictEqual(decoded, input,
+                    "decode() on input with no handles must return original input.");
+                assert.deepStrictEqual(decoded, input,
+                    "input must round-trip through decode(replaceHandles()).");
+
                 const stringified = serializer.stringify(input, handle);
+                // Paranoid check that serializer.stringify() and JSON.stringify() agree.
+                assert.deepStrictEqual(stringified, JSON.stringify(input),
+                    "stringify() of input without handles must produce same result as JSON.stringify().");
+
                 const parsed = serializer.parse(stringified);
                 assert.deepStrictEqual(parsed, input,
-                    "input must round-trip through stringify()/parse().");
+                    "input must round-trip through parse(stringify()).");
 
                 // Paranoid check that serializer.parse() and JSON.parse() agree.
                 assert.deepStrictEqual(parsed, JSON.parse(stringified),
                     "parse() of input without handles must produce same result as JSON.parse().");
             });
         }
 
-        it("replaceHandles() must round-trip undefined", () => {
+        // Non-finite numbers are coerced to null.  Date is coerced to string.
+        const tricky = createNestedCases([-Infinity, NaN, +Infinity, new Date()]);
```

Comment:
It's debatable if these values are truly "serializable" since the coercion is lossy, but at a minimum our behavior should be consistent with JSON...  and I thought adding coverage might help shake out other unexpected bugs with the use of Boolean-ish logic.